src/main/java/com/google/devtools/build/lib/rules/java/JavaOptions.java

```diff
@@ -601,6 +614,10 @@ public ImportDepsCheckingLevelConverter() {
   public FragmentOptions getHost() {
     JavaOptions host = (JavaOptions) getDefault();
 
+    if (useRemoteJdkAsHostJavaBase
+        && "@bazel_tools//tools/jdk:host_jdk".equals(hostJavaBase.toString())) {
```

Comment:
Can the default be `null`? That would avoid the duplication here, and also make it possible to distinguish between 'unset' and 'explicitly set to the default value'.

Reply:
Done.